File: Medium
Medium/1793-minimum-moves-to-make-array-complementary/minimum-moves-to-make-array-complementary.cpp
class Solution{
public:
    int minMoves(vector<int> &nums, int l){
        int n=nums.size();

        vector<int> diff(2*l+2, 0);
        for(int i=0; i<n/2; i++){
            int a=nums[i];
            int b=nums[n-i-1];
            int sum=a+b;

            int low=min(a, b)+1;
            int high=max(a, b)+l;

            diff[2]+=2;
            diff[2*l+1]-=2;

            diff[low]-=1;
            diff[high+1]+=1;

            diff[sum]-=1;
            diff[sum+1]+=1;
        }

        int sum=0;
        int res=INT_MAX;
        for(int i=2; i<=2*l; i++){
            sum += diff[i];
            res=min(res, sum);
        }

        return res;
    }
};

File: Medium
Medium/1793-minimum-moves-to-make-array-complementary/minimum-moves-to-make-array-complementary.py
class Solution:
    def minMoves(self, nums: List[int], l: int) -> int:
        n=len(nums)

        diff=[0]*(2*l+2)
        for i in range(0, n//2):
            a=nums[i]
            b=nums[n-i-1]
            sum=a+b

            low=min(a, b)+1
            high=max(a, b)+l

            diff[2]+=2
            diff[2*l+1]-=2

            diff[low]-=1
            diff[high+1]+=1

            diff[sum]-=1
            diff[sum+1]+=1

        sum=0
        res=float('inf')
        for i in range(2, 2*l+1):
            sum += diff[i]
            res=min(res, sum)

        return res